### ISSUE TITLE
How to partially format the document?

### ISSUE THREAD

**Issue by Robin329** [ayamir/nvimdots#953]
### Feature description

当你选中某一部分代码时，只格式化选中这部分，其他的部分不需要进行代码格式化。

**Comment by ayamir** [comment]
FYI: https://github.com/joechrisellis/lsp-format-modifications.nvim

**Comment by Robin329** [comment]
> FYI: https://github.com/joechrisellis/lsp-format-modifications.nvim

这个插件是指针对修改部分格式化？如果想手动格式化某选中部分是否支持呢？

**Comment by Jint-lzxy** [comment]
`textDocument/rangeFormatting` is supported upstream, but we have no plan to support that in this repo because we don't find rangeFormatting very useful. If you want this feature, you need to add the mapping urself:
```lua
["v|<your-keymap>"] = map_cmd("<Cmd>lua vim.lsp.buf.format()<CR>"):with_noremap():with_silent(),
```
Then use this mapping in visual mode to partially format the document (if the server supports `textDocument/rangeFormatting`).

**Caveat:** Built-in formatting functions does not perform any checks, so use it at ur own risk.

**Comment by Robin329** [comment]
> partially format the document

针对c++/c如何修改呢？c/c++使用的clangd lsp server

**Comment by Jint-lzxy** [comment]
All u need to do is to add a mapping, see [Wiki:Usage#Modify_keymap](https://github.com/ayamir/nvimdots/wiki/Usage#modify-keymap).

**Comment by Robin329** [comment]
`"<Cmd>lua vim.lsp.buf.format()<CR>"`这个是针对所有语言都支持对吗？

**Comment by Jint-lzxy** [comment]
Ys, as long as the corresponding server supports `textDocument/rangeFormatting`.

**Comment by Robin329** [comment]
如选中了这部分code，很明显下面drm_core_exit()的对齐方式不对，因此我使用快捷键但是没有工作。
![image](https://github.com/ayamir/nvimdots/assets/22770951/fa89eb98-c15c-445c-934f-faeaee8a6542)
我在mapping.lua中增加如下快捷键：
	`["v|F"] = map_cmd("<Cmd>lua vim.lsp.buf.format()<CR>"):with_noremap():with_silent():with_desc("edit: Partially format the document"),`

**Comment by Jint-lzxy** [comment]
Some keymaps are reserved in visual mode (like `F`), try `<leader>f` to see if it works?

**Comment by Robin329** [comment]
> Some keymaps are reserved in visual mode (like `F`), try `<leader>f` to see if it works?

可以工作了，感谢。

**Comment by Robin329** [comment]
遇到一个问题，格式化我选中的部分之后，其他没有选中的会受到影响？
![image](https://github.com/ayamir/nvimdots/assets/22770951/2c9a1916-a087-443b-bff1-9bd1aa7a3914)

**Comment by Jint-lzxy** [comment]
> 遇到一个问题，格式化我选中的部分之后，其他没有选中的会受到影响？

Yeah this is the way `clang-format` implement `textDocument/rangeFormatting`. It still needs to ensure that no more errors will be induced b/c of this change.

**Comment by jh244212647** [comment]
有计划接入https://github.com/joechrisellis/lsp-format-modifications.nvim这个插件么，目前Null-ls调用clang-format还都是格式化整个文件，希望能像vscode那样，也可以选择modification

**Comment by ayamir** [comment]
I'm working for it.
